drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -479,10 +480,12 @@ void intel_adsp_hda_dma_isr(void)
 	for (i = 0; i < ARRAY_SIZE(host_dev); i++) {
 		dma_ctx = (struct dma_context *)host_dev[i]->data;
 		cfg = host_dev[i]->config;
-
-		for (j = 0; j < dma_ctx->dma_channels; j++) {
-			if (!atomic_test_bit(dma_ctx->atomic, j))
+		enabled_chs = atomic_get(dma_ctx->atomic);
+		for (j = 0; enabled_chs && j < dma_ctx->dma_channels; j++) {
+			if (!(enabled_chs & BIT(j))) {
 				continue;
+			}
+			enabled_chs &= ~(BIT(j));
```

Comment:
`BIT()` is properly defined, no need for additional parentheses

Reply:
Ok this got merged already, I can fix if we have other changes to do in a follow-up.